File: languages/dotnet/src/provider/MeanAnalysis.cs
using experimento.src.model;
using System.Collections.Generic;

namespace experimento.src.provider {

    /// <summary>
    ///  Classe para analise de dados, obtem a media dos valores dos creditos dos usuarios
    /// </summary>
    public class MeanAnalysis : TableAnalysis<double> {
        
        /// <summary>
        /// Realiza a analise de media dos dados
        /// </summary>
        /// <param name="list">Lista de dados a ser analisada</param>
        /// <returns>Media dos valores dos creditos</returns>
        public double analysis(List<UserInfo> list) {

            double sum = 0;
            foreach(UserInfo userInfo in list) {
                sum += userInfo.credit;
            }

            return sum/list.Count;
        }
    }
}

File: languages/scala/src/provider/MeanAnalysis.scala
package provider

import model.{TableAnalysis, UserInfo}

/**
  * Classe para analise de dados, obtem a media dos valores dos creditos dos usuarios
  * @see model.TableAnalysis
  */
class MeanAnalysis extends TableAnalysis[Double] {

  /**
    * Realiza a analise de media dos dados
    * @param userInfoList Lista de dados a ser analisada
    * @return Media dos valores dos creditos
    * @see TableAnalysis
    */
  override def analysis(userInfoList: List[UserInfo]): Double = {

    var sum: Double = 0.0

    userInfoList.foreach(sum += _.credit)

    return sum/userInfoList.length
  }
}
